networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/ActiveEth2Network.java

```diff
@@ -63,15 +67,20 @@ public ActiveEth2Network(
   }
 
   private void startup() {
-    state.set(State.RUNNING);
+    recentChainData.subscribeBestBlockInitialized(this::initGossipManagers);
+  }
+
+  public void initGossipManagers() {
     BlockValidator blockValidator = new BlockValidator(recentChainData, new StateTransition());
     AttestationSubnetSubscriptions attestationSubnetSubscriptions =
         new AttestationSubnetSubscriptions(discoveryNetwork, recentChainData, eventBus);
-    blockGossipManager = new BlockGossipManager(discoveryNetwork, eventBus, blockValidator);
+    blockGossipManager =
+        new BlockGossipManager(discoveryNetwork, eventBus, blockValidator, recentChainData);
     attestationGossipManager =
         new AttestationGossipManager(eventBus, attestationSubnetSubscriptions);
     aggregateGossipManager =
         new AggregateGossipManager(discoveryNetwork, eventBus, recentChainData);
+    state.set(State.RUNNING);
```

Comment:
I think returning from `start()` before moving to the `RUNNING` state violates the intention of the `start` / `stop` methods.  You'd expect to be able to successfully `stop()` the network after successfully starting it.

Reply:
Reverted that change.